### ISSUE TITLE
[no-shadow] false positive for enum members

### ISSUE THREAD

**Issue by evertbouw** [typescript-eslint/typescript-eslint#2721]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-shadow": "error",
  }
}
```

```ts
enum Foo {
}

enum Bar {
    Foo // 'Foo' is already declared in the upper scope. eslint@typescript-eslint/no-shadow
}
```

**Expected Result**

no error

**Actual Result**

 'Foo' is already declared in the upper scope. 

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.6.0` |
| `@typescript-eslint/parser`        | `4.6.0` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.12.1` |
| `node`                             | `14.15.0` |

**Comment by bradzacher** [comment]
It's a little known fact that enums create their own scope, and enum members create names within that scope.

```ts
const Foo = 2;

enum​ ​Bar​ ​{​
    ​Foo = 1​,
    Baz = Foo,
​}

const result = Bar.Baz;
```

What does result equal? Is it 1 or is it 2? You'd probably be surprised to learn that it is 1, because the enum member shadows the variable name in the outer scope.
